configure.ac

```diff
@@ -1546,6 +1551,35 @@ if test "$enable_threads" = "multiple" ; then
 fi
 AC_DEFINE_UNQUOTED([MPICH_THREAD_GRANULARITY],$thread_granularity,[Method used to implement atomic updates and access])
 
+# Check for enable-vci-method choice
+vci_method=MPICH_VCI__ZERO
+case $enable_vci_method in
+    zero)
+    # Essentially single VCI
+    vci_method=MPICH_VCI__ZERO
+    ;;
+    communicator)
+    # Every communicator gets a new VCI in a round-robin fashion
+    vci_method=MPICH_VCI__COMM
+    ;;
+    tag)
+    # Explicit VCI info embedded with a tag scheme (5 bit src vci, 5 bit dst vci, plus 5 bit user)
+    vci_method=MPICH_VCI__TAG
+    ;;
+    implicit)
+    # An automatic scheme taking into account of user hints
+    vci_method=MPICH_VCI__IMPLICIT
```

Comment:
Isn't `IMPLICIT` a superset of `COMM`. Why not just have `IMPLICIT`?

Reply:
This is a placeholder for the hashing scheme in the current Intel's branch.

Remember that these config options are all experimental options for the time being for development purpose (for exploring purposes). I believe eventually we'll settle down to the explicit scheme.

Reply:
I was suggesting to get rid of `COMM`

Reply:
I also have the same question, do we need `COMM` when we already have `IMPLICIT`?

Reply:
Hypothetically, we could name them `scheme1`, `scheme2`, `...`. Currently, @rzambre 's scheme is different from Intel's implicit scheme. We could add hints and switch "sub"-schemes inside the implicit hashing, that adds complexity -- different author responsible for unrelated schemes are forced to touch the same code, and it adds switching overhead.  Right now, `comm` and `implicit` are just labels for different experimental schemes, I don't think we should hang up too much on it. For production, we probably will always do `implicit`, so for production users, they don't even need see this config option.